Cybersecurity_Tools/DEEPFAKE/extension/backend/main.py

```
@@ -0,0 +1,305 @@
+        prediction = image_model.predict(image, verbose=0)[0][0]
+        confidence = float(prediction) if prediction > 0.5 else float(1 - prediction)
+        label = "Real" if prediction > 0.5 else "Fake"
+        
+        return {
+            "result": label,
+            "confidence": confidence,
+            "error": None
+        }
+    except Exception as e:
+        logger.error(f"Image detection error: {str(e)}")
+        raise HTTPException(status_code=500, detail=f"Image processing failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)